troubleshooting/faq.md

```diff
@@ -161,6 +162,26 @@ So it’s important to understand the different storage driver options available
 
 We recommend that you use the `overlay` driver, rather than `devicemapper`; for help configuring this [please see the following tutorial](https://docs.docker.com/engine/userguide/storagedriver/overlayfs-driver/#configure-docker-with-the-overlayoverlay2-storage-driver)
 
+## What should I do with git dependencies on closed networks?
+
+If you are installing the package that have a dependency on a git url, npm will fetch that package from github itself because git dependencies are valid in npm.
+
+```
+"dependencies" : {
+  "name1" : "git://github.com/user/project.git#commit-ish",
+  "name2" : "git://github.com/user/project.git#commit-ish"
+}
+```
+When you are installing the package in the closed network where a dependency references a git url rather than a package name, you are unable to access due to no internet connectivity.
+
+In this situation you have to download the tarball of package from github on a local and re-publish to the private registry.
```

Comment:
`In this situation, one approach you can take is to download a tarball version of the package and republish it to your private registry.`

Let's add a note about fetching the tarball version of a tag on GitHub, documentation for doing this can be found here, which we can work from:

http://stackoverflow.com/questions/5746325/how-do-i-download-a-tarball-from-github-using-curl